owncloudApp/src/main/java/com/owncloud/android/files/services/CameraUploadsSyncJobService.java

```diff
@@ -182,6 +181,7 @@ private synchronized void handleFile(File localFile) {
 
             if (mOCCameraUploadSync == null) {
                 Log_OC.d(TAG, "There's no timestamp to compare with in database yet, not continue");
+                return;
```

Comment:
without the return, it will crash on L188
`if (isImage && localFile.lastModified() <= mOCCameraUploadSync.getPicturesLastSync()`

btw, the db layer was as expected (to be diplomatic) a surprise

Reply:
> btw, the db layer was as expected (to be diplomatic) a surprise

Please if you consider there's something to improve, put it directly, just complaining is not useful for anyone.